src/Agent.Worker/Release/ReleaseServer.cs

```diff
@@ -5,41 +5,69 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
-
 using Microsoft.VisualStudio.Services.Agent.Util;
-using Microsoft.VisualStudio.Services.Common;
 using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Clients;
 using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Contracts;
 using Microsoft.VisualStudio.Services.WebApi;
 using RMContracts = Microsoft.VisualStudio.Services.ReleaseManagement.WebApi;
 
-namespace Agent.Worker.Release
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
 {
-    public class ReleaseServer
+    [ServiceLocator(Default = typeof(ReleaseServer))]
+    public interface IReleaseServer : IAgentService
+    {
+        Task ConnectAsync(VssConnection jobConnection);
+        IEnumerable<AgentArtifactDefinition> GetReleaseArtifactsFromService(
+            int releaseId,
+            Guid projectId,
+            CancellationToken cancellationToken = default(CancellationToken));
+        Task<RMContracts.Release> UpdateReleaseName(
+            string releaseId,
+            Guid projectId,
+            string releaseName,
+            CancellationToken cancellationToken = default(CancellationToken));
+    }
+    public class ReleaseServer : AgentService, IReleaseServer
     {
         private VssConnection _connection;
-        private Guid _projectId;
 
-        private ReleaseHttpClient _releaseHttpClient { get; }
+        private ReleaseHttpClient _releaseHttpClient;
 
-        public ReleaseServer(VssConnection connection, Guid projectId)
+        public async Task ConnectAsync(VssConnection jobConnection)
         {
-            ArgUtil.NotNull(connection, nameof(connection));
+            _connection = jobConnection;
+            int attemptCount = 5;
+            while (!_connection.HasAuthenticated && attemptCount-- > 0)
```

Comment:
I wonder if this should be added to VssConnection?

Reply:
Perhaps in another PR. I wanted to make as few changes to the product code as possible in this PR. If we want to revisit that, it should be done in a separate PR.